### PR TITLE
[Issue/605][Android]Apply OS font size scale change in editor

### PR DESCRIPTION
**Pull request by hotchemi**
## Issue

- Fixes android part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/605

## Overview

This PR fixes GM editor to apply system level font scale change immediately.

Honestly this change is a bit tricky as React Native has a [mechanism](https://facebook.github.io/react-native/docs/text#allowfontscaling) that allows `<Text>` component to scale font size according to system setting. **But the point is we're using GM component on top of fragment of which [retain flag is true](https://github.com/wordpress-mobile/WordPress-Android/blob/ff8a0692400f02286331f09a4f3705f8b2c54014/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java#L193)**, we can't rely on RN font scaling system.

After several investigations, I've noticed we can detect fontScale change on RN side by observing `Dimensions` change and added a function to calculate new font size to deal with the issue.

### Note

This PR doesn't affect to iOS side since the situation I explained above is a different on each platform.

### Links

- https://facebook.github.io/react-native/docs/text#allowfontscaling
- https://facebook.github.io/react-native/docs/dimensions

### Screenshots

| Before | After |
| ----------- | ----------- |
|||

### Test

- Run example app on Android
- Turn on `Use Block Editor` from app setting
- Open editor screen and type something
- Go to OS setting(Accessibility→Font Size) and change font size
- Back to editor screen and confirm content font size has been changed

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
       <activity
         android:name=".MainActivity"
         android:label="@string/app_name"
-        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
+        android:configChanges="keyboard|keyboardHidden|orientation|screenSize|fontScale"
```
Comment (hotchemi) [inline android/app/src/main/AndroidManifest.xml]: this is not directly related to the issue but to align condition with wp-android added `fontScale` to prevent activity recreation by fontScale change.

File: react-native-aztec/src/utils.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Platform } from 'react-native';
+
+// Basically React Native has a mechanism to scale font size automatically.
+// But GM is rendered on top of fragment of which retain flag is true.
+// Hence we deal with fontScale change manually on our side.
+export function getScaledFontSize( fontSize, fontScale ) {
```
Comment (hotchemi) [inline react-native-aztec/src/utils.js]: currently there's no dependency of unit test in `react-native-aztec` but if ok I'd like to install jest and write an unit test for the function🙇

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (hypest) [reply android/app/src/main/AndroidManifest.xml]: 👋 @hotchemi , does this change make the demo app differ in behavior between Android and iOS?

I'm thinking that making the demo app behave closer to WPAndroid might not be a desirable goal. Instead, the demo app should be as simple as possible and ideally not hide any "mistakes" or "hacks" WPAndroid implements. Also, ideally the Android and iOS implementations of the demo app should be as identical as possible.

File: react-native-aztec/src/utils.js (hunk 2 above)
Comment (hypest) [reply react-native-aztec/src/utils.js]: Let's hold on a bit while we're still assessing the solution in this PR.

While at it though, what do you have in mind for how such a unit test would be run? Will it become part of the Jest testsuite already in place in gutenberg-mobile?

File: react-native-aztec/src/utils.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+	if ( Platform.OS === 'ios' ) {
+		return fontSize ? fontSize : undefined; // to not to affect to iOS
+	}
+	const baseFontSize = 18;
```
Comment (daniloercoli) [inline react-native-aztec/src/utils.js]: Should we use the default value set in RN for this? https://github.com/facebook/react-native/blob/1151c096dab17e5d9a6ac05b61aacecd4305f3db/ReactAndroid/src/main/java/com/facebook/react/uimanager/ViewDefaults.java#L15

Comment (hotchemi) [reply react-native-aztec/src/utils.js]: This is a bit tricky but I did reverse-engineering to find out 18 is the base fontSize for paragraph block. If we use 14 you'll see the base font size is smaller than before.

| no fontSize specify | baseFontSize=18 |
| ----------- | ----------- |
|||

But yea it's too tricky...

**Comment by hotchemi** [comment]
@daniloercoli @hypest Thank you for your review🙇

After opening PR I'm kind of wondering this is too workaround to merge to upstream..🤔Yes we can mimic plausible behavior with this change but I'm kind of skeptical this change is maintainable enough for team.

To resolve the issue fundamentally I think we should remove `setRetainInstance(true)` from GutenbergContainerFragment(and prepare something for screen orientation in other way) but what do you think?🤔If the issue is really crucial for user we need a quick workaround but seems it's not in this case. Or let me know if there're other ideas to deal with the issue🙇

**Comment by hotchemi** [comment]
Seems like this PR is not a fundamental solution to address the issue, let me close for the time being 👋
